master/internal/command/command.go

```diff
@@ -38,460 +37,243 @@ import (
-		}
-		ctx.Respond(&apiv1.KillCommandResponse{Command: c.toCommand(ctx)})
-
-	case *apiv1.SetCommandPriorityRequest:
-		err := c.setPriority(int(msg.Priority), true)
-		if err != nil {
-			ctx.Respond(err)
-			return nil
-		}
-		ctx.Respond(&apiv1.SetCommandPriorityResponse{Command: c.toCommand(ctx)})
-
-	case *shellv1.Shell:
-		ctx.Respond(c.toShell(ctx))
-
-	case *apiv1.GetShellRequest:
-		ctx.Respond(&apiv1.GetShellResponse{
-			Shell:  c.toShell(ctx),
-			Config: protoutils.ToStruct(c.Config),
-		})
-
-	case *apiv1.KillShellRequest:
-		err := task.DefaultService.Signal(c.allocationID, task.KillAllocation, "user requested kill")
-		if err != nil {
-			ctx.Log().WithError(err).Warn("failed to kill allocation")
-		}
-		ctx.Respond(&apiv1.KillShellResponse{Shell: c.toShell(ctx)})
-
-	case *apiv1.SetShellPriorityRequest:
-		err := c.setPriority(int(msg.Priority), true)
-		if err != nil {
-			ctx.Respond(err)
-			return nil
-		}
-		ctx.Respond(&apiv1.SetShellPriorityResponse{Shell: c.toShell(ctx)})
-
-	case *tensorboardv1.Tensorboard:
-		ctx.Respond(c.toTensorboard(ctx))
-
-	case *apiv1.GetTensorboardRequest:
-		ctx.Respond(&apiv1.GetTensorboardResponse{
-			Tensorboard: c.toTensorboard(ctx),
-			Config:      protoutils.ToStruct(c.Config),
-		})
-
-	case *apiv1.KillTensorboardRequest:
-		err := task.DefaultService.Signal(c.allocationID, task.KillAllocation, "user requested kill")
-		if err != nil {
-			ctx.Log().WithError(err).Warn("failed to kill allocation")
-		}
-		ctx.Respond(&apiv1.KillTensorboardResponse{Tensorboard: c.toTensorboard(ctx)})
-
-	case *apiv1.SetTensorboardPriorityRequest:
-		err := c.setPriority(int(msg.Priority), true)
-		if err != nil {
-			ctx.Respond(err)
-			return nil
-		}
-		ctx.Respond(&apiv1.SetTensorboardPriorityResponse{Tensorboard: c.toTensorboard(ctx)})
-
-	case *apiv1.DeleteWorkspaceRequest:
-		if c.Metadata.WorkspaceID == model.AccessScopeID(msg.Id) {
-			err := task.DefaultService.Signal(
-				c.allocationID,
-				task.KillAllocation,
-				"user requested workspace delete",
-			)
-			if err != nil {
-				ctx.Log().WithError(err).Warn("failed to kill allocation while deleting workspace")
-			}
-		}
-
-	case sproto.ContainerLog:
-
-	case terminateForGC:
-		ctx.Self().Stop()
-
-	case sproto.SetGroupWeight:
-		err := c.SetWeight(msg.Weight)
-		if err != nil {
-			ctx.Log().WithError(err).Info("setting command job weight")
-		}
-		if ctx.ExpectingResponse() {
-			ctx.Respond(err)
-		}
-
-	case sproto.SetGroupPriority:
-		err := c.setPriority(msg.Priority, true)
-		if err != nil {
-			ctx.Log().WithError(err).Info("setting command job priority")
-		}
-		if ctx.ExpectingResponse() {
-			ctx.Respond(err)
+			c.syslog.WithError(err).Error("marking task complete")
 		}
-
-	default:
-		return actor.ErrUnexpectedMessage(ctx)
 	}
-	return nil
-}
 
-func (c *command) SetPriority(priority int, forward bool) error {
-	c.mu.Lock()
-	defer c.mu.Unlock()
+	jobservice.DefaultService.UnregisterJob(c.jobID)
```

Comment:
we need this, right? else we are susceptible to a lock cycle
```suggestion
	go jobservice.DefaultService.UnregisterJob(c.jobID)
```

Reply:
yeah that should have the `go`, not sure at what point I removed that, but it should be there